### PR TITLE
Add post stop command for cleanup

### PR DESCRIPTION
**Pull request by crosbymichael**
For use in the unit file like

`ExecStopPost=/usr/local/bin/dockerd post-stop`

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

### CODE REVIEW

**Comment by andrewhsu** [comment]
LGTM wrt cleaning up stuff

File: config.go [hunk 1]
```diff
@@ -45,18 +45,13 @@ type Config struct {
...
-		case containerScope == c.Scope:
-			return true
-		default:
+		if containerScope == AnyScope && c.Scope != containerScope {
 			return false
 		}
+		if containerScope == "" {
+			return true
+		}
+		return c.Scope == AnyScope
```
Comment (dhiltgen) [inline config.go:54]: if we ignore the CLI `update` flow this works fine, and that flow can be mitigated by making changes to the CLI - in either case, I think this incrementally gets use closer to a good end state (or maybe the final state) so this part lgtm.

File: cleanup.go [hunk 2]
```diff
@@ -2,27 +2,66 @@ package main
...
+		task.Kill(ctx, unix.SIGKILL)
+		_, err = task.Delete(ctx)
 		return err
 	}
-	if status.Status != containerd.Stopped {
-		return errors.New("unable to start running container")
+	if err := task.Kill(ctx, unix.SIGKILL); err != nil {
+		fmt.Fprintf(os.Stderr, err.Error())
 	}
+	<-wait
```
Comment (dhiltgen) [inline cleanup.go:54]: 'nit - perhaps it's pedantic, but wondering if this could be hardened with:
```
select {
case <- wait
case <- ctx.Done():
}
```

**Review by dhiltgen — APPROVED** [review]

File: cleanup.go (hunk 2 above)
Comment (crosbymichael) [reply cleanup.go:54]: actually, since we pass the context to Wait() it will automatically close the connection as well as the channel on a timeout

**Pull request merged**